src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2887 @@
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Declare variables.
+!-----------------------------------------------------------------------
+  REAL(r_kind),INTENT(IN   ) :: N0,alpha,Ai,Bi,Ci,alp_a,alp_b,beta_a,beta_b
+  REAL(r_kind),INTENT(IN   ) :: lamda
+
+  REAL(r_kind),INTENT(  OUT) :: refIceHH,refIceVV
+
+  !local variables
+  REAL(r_kind) :: gamma_h, gamma_v, expon_h, expon_v
+  REAL(r_kind) :: N0_units
+  REAL(r_double) :: gamma
+
+  gamma_h = sngl(gamma(dble(alpha) + 2.0_r_double*dble(beta_a)+1.0_r_double))
+  expon_h = -(alpha+2*beta_a+1)
+  gamma_v = sngl(gamma(dble(alpha) + 2.0_r_double*dble(beta_b)+1.0_r_double))
+  expon_v = -(alpha + 2*beta_b+1)
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+  N0_units = (1.e-3_r_kind)**(4.0_r_kind+alpha)
+
+  refIceHH = mm3toDBZ*radar_Const*gamma_h*(N0*N0_units)*                     &
+              (Ai*alp_a**2+Bi*alp_b**2+2*Ci*alp_a*alp_b)*                 &
+             (lamda*1.e-3_r_kind)**expon_h
+
+  refIceVV = mm3toDBZ*radar_Const*gamma_v*(N0*N0_units)*                    &
+             (Bi*alp_a**2+Ai*alp_b**2+2*Ci*alp_a*alp_b)*                  &
+             (lamda*1.e-3_r_kind)**expon_v
+
+
+END SUBROUTINE partialRefIce
+
+SUBROUTINE partialRhoIce(N0,alpha,Ci,Di,alp_a,alp_b,beta_a,beta_b,rho_0,lamda,refIceHV)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! This function calculates the cross components, Z_hv, for each species
+! for rho_hv calculation.
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 12/16/2007
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Declare variables.
+!-----------------------------------------------------------------------
+  REAL(r_kind),INTENT(IN   ) :: N0,alpha,Ci,Di,alp_a,alp_b,beta_a,beta_b
+  REAL(r_kind),INTENT(IN   ) :: rho_0,lamda
+
+  REAL(r_kind),INTENT(  OUT) :: refIceHV
+
+  !local variables
+   REAL(r_kind) :: gamma_hv, expon
+   REAL(r_kind) :: N0_units
+   REAL(r_double) :: gamma
+
+   gamma_hv = sngl(gamma(dble(beta_a)+dble(beta_b)+dble(alpha) + 1.0_r_double))
+   expon = -(alpha + beta_a + beta_b + 1)
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+   N0_units = (1.e-3_r_kind)**(4.0_r_kind+alpha)
+
+   refIceHV = mm3todBZ*radar_Const*gamma_hv*(N0*N0_units)*             &
+               (Ci*alp_a**2+Ci*alp_b**2+2*Di*alp_a*alp_b*rho_0)*    &
```

Comment:
Please add _r_kind to 2
`(i.e., (Ci*alp_a**2+Ci*alp_b**2+2_r_kind*Di*alp_a*alp_b*rho_0)`